test/test_nn.py

```diff
@@ -2071,23 +2071,46 @@ def test_embedding_dense_grad_cuda(self):
         self._test_embedding_dense_grad("cuda")
 
     def test_embedding_sparse_backward(self):
+        self._test_embedding_backward(False)
+
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    def test_embedding_sparse_half_backward(self):
+        # same as test_embedding_sparse_backward above but testing half types in
+        # cuda. cpu sum not supported for half types.
+        self._test_embedding_backward(True)
+
+    def _test_embedding_backward(self, half_cuda=True):
         embedding = nn.Embedding(10, 3, sparse=True)
+        tensor = torch.tensor([[7, 1, 3]])
+        ones = torch.tensor(1.).expand(3, 3).half()
+        tensorTwice = torch.tensor([[7, 1, 3, 7, 1, 3]])
```

Comment:
nit: you can probably use `torch.repeat`.

Reply:
updated to use repeat